### PR TITLE
[grpc] Conditionally depend on  "abseil"

### PR DESCRIPTION
**Pull request by redbaron**
Pull "abseil" dependency only when "absl-sync" feature is enabled.

**Describe the pull request**

- #### What does your PR fix?  

  Makes abseil dependency conditional on "absl-sync" feature

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  I am still working on this PR

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@redbaron ,Thanks for your pr, Is this PR being worked on?

File: ports/grpc/vcpkg.json [hunk 1]
```diff
@@ -36,7 +35,8 @@
   ],
   "features": {
     "absl-sync": {
-      "description": "Use abseil synchronization module"
+      "description": "Use abseil synchronization module",
+      "dependencies": [ "abseil" ],
```
Comment (redbaron) [inline ports/grpc/vcpkg.json:39]:
```suggestion
      "dependencies": [ "abseil" ]
```

**Comment by JonLiu1993** [comment]
Ping @redbaron for response, Is this  pr still work on?

**Comment by JonLiu1993** [comment]
@redbaron, Could you please sign in the CLA, I will help you to complete this pr

**Comment by redbaron** [comment]
@JonLiu1993 , done

**Comment by JonLiu1993** [comment]
I will try to fix this error: 
````
CMake Error at C:/a/1/s/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
  Could not find a package configuration file provided by "absl" with any of
  the following names:

    abslConfig.cmake
    absl-config.cmake

  Add the installation prefix of "absl" to CMAKE_PREFIX_PATH or set
  "absl_DIR" to a directory containing one of the above files.  If "absl"
  provides a separate development package or SDK, be sure it has been
  installed.
````

**Comment by JonLiu1993** [comment]
I will close this pr first, I will reopen this pr later when I have time to solve it

**Pull request closed without merging**